velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -217,7 +218,20 @@ dwio::common::StripeProgress getStripeProgress(
  *
  * This method assumes each input `ColumnarBatch` have same schema.
  */
-void Writer::write(const VectorPtr& data) {
+void Writer::write(const VectorPtr& dataToWrite) {
+  VectorPtr data = dataToWrite;
+  if (data && schema_) {
+    auto rowType = std::dynamic_pointer_cast<const velox::RowType>(schema_);
+    auto rowVector =
+        std::dynamic_pointer_cast<facebook::velox::RowVector>(data);
+    data = std::make_shared<facebook::velox::RowVector>(
```

Comment:
Can you add check that schema_ is equivalent to data->type()?

Reply:
@mbasmanova Updated.